about_matching.exs

```diff
@@ -33,6 +33,20 @@ defmodule AboutMatching do
     assert_raise MatchError, fn -> [a , a] = [1 , __?] end
   end
 
+  think "Matched values overwrite variable values" do
+    a = 1
+    assert a == __?
+
+    [a, _] = [2, 3]
+    assert a == __?
+  end
+
+  think "Values can be pinned to prevent them from being overwritten" do
```

Comment:
It's kind of like when you do `{:ok, result} = some_func`; you want it to fail if the first item wasn't `:ok`.